go/vt/sqlparser/parsed_query_test.go

```diff
@@ -203,3 +203,38 @@ func TestParseAndBind(t *testing.T) {
 		})
 	}
 }
+
+func TestParseAndBindWithQualifier(t *testing.T) {
+	testcases := []struct {
+		in        string
+		qualifier string
+		binds     []*querypb.BindVariable
+		out       string
+	}{
+		{
+			in:        "drop table `%s`",
+			qualifier: "tbl",
+			out:       "drop table `tbl`",
+		},
```

Comment:
We should test w/o escaping in the in query too. Both when escaping is required (e.g. `test-name`) and when it's not. I think it's problematic to include the escaping in the incoming query, as it's up to the parser to provide the escaping when needed. I see specifying the escaping for the identifier in the raw query string as being equivalent to specifying quotes when we know the bind placeholder is for a string: `'%a'`. I suspect that when the escaping IS needed, we'll have double escaping here:
```
drop table ``test-name``
```

Reply:
Yeah, I feel like we should get @vitessio/query-serving 's official opinion. No strong opinions on my side. I understand both approaches, they both make sense.